applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationCredentialsPolicy.java

```diff
@@ -90,28 +81,20 @@
                         return Mono.error(new NoSuchAlgorithmException("Unable to locate SHA-256 algorithm."));
                     }
 
-                    final Map<String, String> mapped = getDefaultHeaders(context.httpRequest().headers());
+                    final Map<String, String> mapped = getDefaultHeaders(context.httpRequest().url(), context.httpRequest().headers());
                     final String contentHash = Base64.getEncoder().encodeToString(messageDigest.digest());
 
                     mapped.put(CONTENT_HASH_HEADER, contentHash);
-
```

Comment:
In process method, we don't have to wait till `flatmap` to throw `NoSuchAlgorithmException`, we can use `Exceptions.propagate` in collect.supplier, like below. Also instead of creating new `NoSuchAlgorithmException` it will be good to rethrow it as it is.

```java
@Override
public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
    final Flux<ByteBuf> contents = context.httpRequest().body() == null
            ? Flux.just(getEmptyBuffer())
            : context.httpRequest().body().defaultIfEmpty(getEmptyBuffer());
    //
    return contents.collect(() -> {
        try {
            return MessageDigest.getInstance("SHA-256");
        } catch (NoSuchAlgorithmException e) {
            throw Exceptions.propagate(e);
        }
    }, (messageDigest, byteBuf) -> messageDigest.update(byteBuf.nioBuffer())).flatMap(messageDigest -> {
        final Map<String, String> mapped = getDefaultHeaders(context.httpRequest().url(), context.httpRequest().headers());
        final String contentHash = Base64.getEncoder().encodeToString(messageDigest.digest());

        mapped.put(CONTENT_HASH_HEADER, contentHash);
        mapped.forEach((key, value) -> context.httpRequest().headers().set(key, value));

        return next.process().doOnSuccess(this::logResponseDelegate);
    });
}
```

Reply:
Thanks! I did not know that. Super helpful. Fixed.